src/SubPool.sol

```diff
@@ -0,0 +1,157 @@
+        if (msg.sender != address(factory)) {
+            revert SubPool__OnlyFactory();
+        }
+        _;
+    }
+
+    /// @notice Signal the intent to announce exit the solver pool.
+    function announceExit() external auth {
+        factory.announceExit();
+    }
+
+    /// @notice Execute the exit after the exit delay has elapsed.
+    function exit() external auth {
+        uint256 exitTimestamp = factory.exitTimestamp(address(this));
+        // can skip the 0 check i.e. pool not announced an exit check because block.timestamp
+        // will never be < 0
+        if (block.timestamp < exitTimestamp) revert SubPool__ExitDelayNotElapsedYet();
+
+        uint256 collateralBalance = ERC20(collateralToken).balanceOf(address(this));
+        uint256 cowBalance = ERC20(COW).balanceOf(address(this));
+        SafeTransferLib.safeTransfer(collateralToken, msg.sender, collateralBalance);
+        SafeTransferLib.safeTransfer(COW, msg.sender, cowBalance);
+        SafeTransferLib.safeTransferETH(msg.sender, address(this).balance);
+
+        factory.exitPool();
+    }
+
+    /// @notice withdraw arbitrary tokens
+    /// @dev Can only withdraw non cow and non collateral tokens while the pool is active or in exit delay.
+    ///      When the pool's exit delay has elapsed it can withdraw any token and ether balance.
+    function withdrawTokens(address[] calldata tokens) external auth {
+        uint256 exitTimestamp = factory.exitTimestamp(address(this));
+        bool exitElapsed = exitTimestamp != 0 && block.timestamp >= exitTimestamp;
+
+        if (exitElapsed) {
+            for (uint256 i = 0; i < tokens.length;) {
+                SafeTransferLib.safeTransfer(tokens[i], msg.sender, ERC20(tokens[i]).balanceOf(address(this)));
+                unchecked {
+                    ++i;
+                }
+            }
+            uint256 ethBalance = address(this).balance;
+            if (ethBalance > 0) {
+                SafeTransferLib.safeTransferETH(msg.sender, address(this).balance);
+            }
+        } else {
+            for (uint256 i = 0; i < tokens.length;) {
+                address token = tokens[i];
+                if (token == COW || token == collateralToken) revert SubPool__InvalidWithdraw();
+                SafeTransferLib.safeTransfer(token, msg.sender, ERC20(token).balanceOf(address(this)));
+                unchecked {
+                    ++i;
+                }
+            }
+        }
+    }
+
+    /// @notice Pull required number of tokens from the sender to push the pool
+    ///         above collateralization.
+    function heal() external payable {
+        uint256 amt = collateralDue;
+        uint256 cowAmt = cowDue;
+        uint256 ethAmt = ethDue;
+        payDues(amt, cowAmt, ethAmt);
+    }
+
+    /// @notice Pay partial or full dues.
+    function payDues(uint256 amt, uint256 cowAmt, uint256 ethAmt) public payable {
+        if (msg.value != ethAmt) revert SubPool__InsufficientETH();
```

Comment:
nit:`ethAmt` seems unnecessary in this function (can directly use msg.value).

Reply:
It is a parameter, because it is also used in `heal` which is basically fully paying all bill dues, so it needs to specify the eth amt

Reply:
will remove the partial due payments